### DISCUSSION TITLE
Proportionally randomizing experimental block sequence

### DISCUSSION THREAD

**Discussion by j22tis** [jspsych/jsPsych#1780] [Q&A] [answered]
Hi,

I need some advice on constructing a randomization algorithm for a sequence of blocks in my experiment.

I have two types of blocks - hard and easy, and I want to generate a random sequence of 26 of the blocks, but to keep the final proportion of blocks 50/50 or at least close to it, e.g 45/ 55.

I currently used the sampleWithReplacement approach `var experimental_timeline = jsPsych.randomization.sampleWithReplacement(blocks, 26);,` but the final proportion of hard/easy blocks comes sometimes out too unbalanced (e.g 8 and 18), so I'm wondering if there is some other way to approach this?

**Comment by becky-gilbert** [comment]
One approach would be to create one array with all 26 blocks (13 of each), and then use `jsPsych.randomization.shuffle` to randomize the block order. Would that work?

**Comment by jodeleeuw** [comment, marked as answer]
Check out the randomization.repeat method: https://www.jspsych.org/core_library/jspsych-randomization/#jspsychrandomizationrepeat

**Comment by j22tis** [comment]
Thanks a lot! I tried the randomization repeat method, since it takes less effort and it did the job :)
